### PR TITLE
Docs: Rework of Contributing.md

### PR DESCRIPTION
**Pull request by ReverM**
## What is this fixing or adding?
Adding a lot of details to the contributing.md document to make it more beginner-friendly

## How was this tested?
Feedback?

### CODE REVIEW

**Review by ScipioWright — APPROVED** [review]

**Comment by ReverM** [comment]
To note: I am planning on adding a "quick guide" for tests. It will contain an overview of the most useful tests for logic, an overview of what is required to be tested by a core contributor, as well as a quick overview for webhost testing.

File: docs/contributing.md
Comment (black-sliver) [inline docs/contributing.md:17]: To make the picture appear on a new line
```suggestion
You can turn them on here:  
```

File: docs/contributing.md
Comment (black-sliver) [inline docs/contributing.md:21]: The website part should probably be a bullet point above. Maybe merged with other sub-docs as in

* **Check available documentation**
  * For the website see ...
  * For adding a game see ..., ... and ...

If you think that actually reduces readability, then the website "extras" should be separated out the same way that we reference world docs.

File: docs/contributing.md
Comment (black-sliver) [inline docs/contributing.md:9]:
```suggestion
* **Ensure that all changes are covered by unit tests.** 
It is strongly recommended that unit tests are used to avoid regression and to ensure everything is still working.
```
Imo those two sentences are confusing when combined as they say different things about the same thing.
Maybe "Ensure that **critical** changes are covered by unit tests." Worlds currently can do whatever, basically, and so does javascript because we have no test framework for that.

File: docs/contributing.md
Comment (ReverM) [reply docs/contributing.md:21]: I think I will separate it on a new sentence, thanks for noting it.

Comment (ReverM) [reply docs/contributing.md]: Done

File: docs/contributing.md
Comment (black-sliver) [inline docs/contributing.md:6]: This can be made a shorter and clearer, also indentation
```suggestion
  Please take a look at the [code style documentation](/docs/style.md)
  to ensure uniformity and ease of communication.
```

**Review by black-sliver — COMMENTED** [review]
The other bullet points are also missing proper indentation.

**Comment by nicholassaylor** [comment]
> To note: I am planning on adding a "quick guide" for tests. It will contain an overview of the most useful tests for logic, an overview of what is required to be tested by a core contributor, as well as a quick overview for webhost testing.

This sounds like a great addition that a lot of people will find very useful. I'm suprised that we don't already have something like this.

File: docs/contributing.md
Comment (MatthewMarinets) [inline docs/contributing.md:6]: Minor: This text is indented by two spaces when the following bullet point aren't. Note in either case, Markdown will combine the following lines with the above bullet point into one paragraph / list item. I assume this is intentional.

File: docs/contributing.md
Comment (MatthewMarinets) [inline docs/contributing.md:30]: "please make sure to read the responsibilities as a world maintainer" is a little awkward. Consider one of:
* please make sure to read the responsibilities **of** a world maintainer
* please make sure to read **your** responsibilities as a world maintainer

**Review by MatthewMarinets — COMMENTED** [review]
Quite good overall. One minor grammar nitpick.

File: docs/contributing.md
Comment (black-sliver) [reply docs/contributing.md:6]: yeah, this what i meant by

> The other bullet points are also missing proper indentation.

The other listings should be fixed as well.

Comment (ScipioWright) [inline docs/contributing.md:6]:
```suggestion
  Please take a look at the [code style documentation](/docs/style.md)
  to ensure ease of communication and uniformity.
```

I don't know why this is standing out to me so much now, but it looks really weird with "ease of communication" second.

File: docs/contributing.md
Comment (ScipioWright) [inline docs/contributing.md:23]:
```suggestion

For contribution to the website, please refer to the [WebHost README](/WebHostLib/README.md).

```

File: docs/contributing.md
Comment (ScipioWright) [reply docs/contributing.md:30]: "please make sure to read the [responsibilities as a world maintainer documentation]"

I think that would be the most clear here.

**Review by ScipioWright — CHANGES_REQUESTED** [review]
Some nitpicks, mostly

File: docs/contributing.md
Comment (ReverM) [reply docs/contributing.md:30]: I will consider those changes

Comment (black-sliver) [reply docs/contributing.md:30]: anything?

**Review by black-sliver — APPROVED** [review]
Lgtm. The world test link should probably change once we merge #2348

**Pull request merged**
